src/storage/conversion/ContentTypeReplacer.ts

```diff
@@ -0,0 +1,62 @@
+import type { Representation } from '../../ldp/representation/Representation';
+import { RepresentationMetadata } from '../../ldp/representation/RepresentationMetadata';
+import type { ValuePreferences } from '../../ldp/representation/RepresentationPreferences';
+import { CONTENT_TYPE } from '../../util/Vocabularies';
+import { matchingMediaTypes } from './ConversionUtil';
+import type { RepresentationConverterArgs } from './RepresentationConverter';
+import { RepresentationConverter } from './RepresentationConverter';
+
+/**
+ * A {@link RepresentationConverter} that changes the content type
+ * but does not alter the representation.
+ *
+ * Useful for when a content type is binary-compatible with another one;
+ * for instance, all JSON-LD files are valid JSON files.
+ */
+export class ContentTypeReplacer extends RepresentationConverter {
+  private readonly replacements: Record<string, ValuePreferences> = {};
+
+  /**
+   * @param replacements - Map of content types and content types to replace them by.
+   */
+  public constructor(replacements: Record<string, string>);
+  public constructor(replacements: Record<string, Iterable<string>>);
+  public constructor(replacements: Record<string, any>) {
+    super();
+    for (const [ inputType, outputTypes ] of Object.entries(replacements)) {
+      this.replacements[inputType] = Object.fromEntries(
+        (typeof outputTypes === 'string' ? [ outputTypes ] : [ ...outputTypes ])
+          .map((outputType): [string, number] => [ outputType, 1 ]),
+      );
+    }
+  }
+
+  public async canHandle({ representation, preferences }: RepresentationConverterArgs): Promise<void> {
+    this.getReplacementType(representation.metadata.contentType, preferences.type);
+  }
+
+  /**
+   * Changes the content type on the representation.
+   */
+  public async handle({ representation, preferences }: RepresentationConverterArgs): Promise<Representation> {
+    const contentType = this.getReplacementType(representation.metadata.contentType, preferences.type);
+    const metadata = new RepresentationMetadata(representation.metadata, { [CONTENT_TYPE]: contentType });
+    return { ...representation, metadata };
+  }
+
+  public async handleSafe(args: RepresentationConverterArgs): Promise<Representation> {
+    return this.handle(args);
+  }
+
+  /**
+   * Find a replacement content type that matches the preferences,
+   * or throws an error if none was found.
+   */
+  private getReplacementType(contentType = 'unknown', typePreferences: ValuePreferences = {}): string {
```

Comment:
Just a thought, but would it be interesting to detect potential chains here? E.g. if A->B is valid and B->C is valid A->C is also valid. Might make the configuration easier.

Reply:
Good idea. Would do this in the constructor already though.